packages/firestore/exp/src/api/reference.ts

```diff
@@ -280,6 +278,159 @@ export function addDoc<T>(
     .then(() => docRef);
 }
 
+// TODO(firestorexp): Make sure these overloads are testing via the Firestore
+// integration tests
+export function onSnapshot<T>(
+  reference: firestore.DocumentReference<T>,
+  observer: {
+    next?: (snapshot: firestore.DocumentSnapshot<T>) => void;
+    error?: (error: firestore.FirestoreError) => void;
+    complete?: () => void;
+  }
```

Comment:
I've been meaning to ask this for some time but what's the history behind using 3 separate callbacks in web vs what we do in other languages?

Reply:
They are meant to make it easy to plug in RxObservables.